File: python/2294.partition-array-such-that-maximum-difference-is-k.py
class Solution:
    def partitionArray(self, nums: list[int], k: int) -> int:
        if k == 0:
            return len(set(nums))

        n = len(nums)
        count = 1
        j = 0

        nums.sort()

        for i in range(n):
            if nums[i] - nums[j] <= k:
                continue

            count += 1
            j = i

        return count

File: cpp/2294.partition-array-such-that-maximum-difference-is-k.cpp
#include <vector>
using std::vector;

class Solution {
public:
  int partitionArray(vector<int> &nums, int k) {
    const int n = nums.size();
    int count = 1;

    std::sort(nums.begin(), nums.end());

    for (int i = 0, j = 0; i < n; i++) {
      if (nums[i] - nums[j] <= k) {
        continue;
      }

      count++;
      j = i;
    }

    return count;
  }
};